README.md

```diff
@@ -1,2 +1,228 @@
+    Zephyr CI system through `module.yml`.
+
+- Maximize the amount of code automatically verifiable through Zephyr CI.
+
+- Minimize risk of conflicts in global namespaces through use of the
+  module abbreviation.
+
+  - Global namespaces exist in many areas, including:
+    Kconfig, CMake, C preprocessor symbols; include file pathnames;
+    test and sample names; shell commands; settings identifiers.
+    Using the module abbreviation as a global symbol prefix reduces
+    the chance of collisions as other modules are created.
+
+## Example vs Sample
+
+In Zephyr *example* and *sample* are not treated as synonyms.s
+The site [differencebetween.net][5] describes the difference as:
+
+> Examples are used to exemplify and illustrate something.
+> “Example” is also utilized as a tool for the explanation and
+> reinforcement of a particular point. Moreover, examples are
+> used for strict compliance or as a premeditated experience.
+> In this manner, it is expected that the example will be
+> followed and replicated among its audience
+> 
+> ...
+> 
+> [A] sample is a small part of something much bigger. ... 
+> Samples are often used to describe the quality or nature of a specific whole.
+
+This *example* repository is intended to contain complete
+implementations of simplified extensions (applications, subsystems, etc.).
+Some of the extensions will be *example* projects and tests of
+functionality provided in this module.
+
+Some of the extensions will be *samples* of API usage which ignore
+return values to show the nature of the API usage as simply as possible.
+
+
+# Checking Out Code
+
+The Zephyr module ZXA may be cloned and added to the Zephyr build system
+in exactly the same way as any other repository.
+
+Follow this guide to:
+
+- Get the source code with Zephyr module ZXA as the `west`
+  manifest repository
+- Build and run a sample application
+
+
+## Get Zephyr and install Python dependencies
+
+Clone Zephyr module ZXA and its dependencies into a new `west`
+workspace named :file:`zp`. You'll also install additional Python
+dependencies.
+
+1. Get the Zephyr source code:
+
+      ```bash
+      west init \
+        -m https://github.com/zephyrproject-rtos/example-application \
+        --mr main \
+        --mf zephyr/west.yml \
+        ~/zxa
+      cd ~/zxa
+      west update
+      source zephyr/zephyr-env.sh
+      ```
+
+1. Export a `Zephyr CMake package`. This allows CMake to
+   automatically load boilerplate code required for building Zephyr
+   applications.
+
+      ```bash
+      west zephyr-export
+      ```
+
+1. Install additional Python dependencies specified by any module
+
+      ```bash
+      pip3 install --user -r ~/zxa/zephyr/scripts/requirements.txt
+      ```
+
+1. Configure `west` to place build output in 
+   project-board-specific folders
+
+      ```bash
+      west config \
+        build.dir-fmt ${ZEPHYR_BASE}/../build/{source_dir}/{board}
```

Comment:
This one feels way too opinionated to belong here.

Reply:
Since this repository is intentionally structured to support multiple application projects, we ought to guide users through setting up an environment that maintains artifacts of builds of multiple applications. Using a `west config` setting following an example usage from `west` is no more opinionated than using `west` itself.

Reply:
This type of configuration breaks the usual recommended `west build` + `west flash` workflow. It is also more opaque about where the build directory went. I think configuring a tool a particular way is inherently more opinionated than choosing to use it at all, FWIW.

Reply:
I have used this configuration for months and I don't see how it breaks that workflow. What it does mean is I must be more explicit about which project and/or board I am working with (e.g. native_posix vs a target board, test_a vs test_b).  All very natural and consistent when working with subsystems and modules.

I don't think of it as opaque, but as automatically consistent.  Every command line build in that workspace behaves predictably, so referencing it is almost mindless.  Remember, this is an example, designed to help new users to get up and running quickly.  Those that prefer to not can ignore it.  Also, remember that this is a test case, and we want to make sure that the Zephyr build system supports this configuration.